### DISCUSSION TITLE
$SD.logMessage() not writing to log file

### DISCUSSION THREAD

**Discussion by b263** [elgatosf/elgato#94] [Help]
I'm using Stream Deck version 6.5.0 on macOS.

I want to write to the log file with the JavaScript SDK and $SD.logMessage().

A log file is created with my plugins ID, but it remains empty.

Also a note on the documentation:
The log directory stated in https://docs.elgato.com/sdk/plugins/events-sent#logmessage is wrong. The correct path is: `~/Library/Logs/ElgatoStreamDeck/`.

**Comment by jonasermert** [comment]
Does you $SD.logMessage() function look like this: 

```
$SD.on('someEvent', (context, payload) => {
    // your logic code
    $SD.logMessage("Event received: " + JSON.stringify(payload));
});
```

**Comment by b263** [comment]
I tried calling it on the top level, which seems to be the issue. But there are no requirements documented.

**Comment by ChekTek** [comment]
Hi @b263,

The Stream Deck application writes the log files, so you have to ensure that the plugin is first connected to Stream Deck. You can use the `$SD.onConnected` event listener to test this out: https://github.com/elgatosf/streamdeck-plugin-template?tab=readme-ov-file#start-coding

This is subject to change with the release of our new Node.js SDK and CLI tool, which are currently in beta and can be found here: 

https://www.npmjs.com/package/@elgato/cli
https://www.npmjs.com/package/@elgato/streamdeck

Note that the docs are not yet updated with information related to the beta tools.

**Comment by pixelated-au** [comment]
I can confirm that the documentation for Mac is indeed **_incorrect._** I spent way too long looking for `~/Library/Logs/StreamDeck/`

@ChekTek, can someone please update the documentation on https://docs.elgato.com/sdk/plugins/events-sent#logmessage to be the correct path?

The path should be `~/Library/Logs/ElgatoStreamDeck/`

**Reply by ChekTek** [reply]
Thanks for pointing this out! I've updated the documentation. 🚀
